components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/ActionSecretProcessor.java

```diff
@@ -0,0 +1,229 @@
+                if (isSecretPropertyExists(secretName)) {
+                    ActionMgtServiceComponentHolder.getInstance().getSecretManager()
+                            .deleteSecret(IDN_SECRET_TYPE_ACTION_SECRETS, secretName);
+                }
+            }
+        }
+    }
+
+    public List<AuthProperty> getPropertiesWithSecretReferences(List<AuthProperty> authProperties, String actionId,
+                                                                String authType) throws SecretManagementException {
+
+        List<AuthProperty> referenceUpdatedProperties = new ArrayList<>();
+        for (AuthProperty prop : authProperties) {
+            if (!prop.getIsConfidential()) {
+                referenceUpdatedProperties.add(prop);
+            } else {
+                referenceUpdatedProperties.add(new AuthProperty.AuthPropertyBuilder()
+                        .name(prop.getName())
+                        .isConfidential(prop.getIsConfidential())
+                        .value(buildSecretReference(buildSecretName(actionId, authType, prop.getName()))).build());
+            }
+        }
+
+        return referenceUpdatedProperties;
+    }
+
+    /**
+     * Encrypt secret property.
+     *
+     * @param authProperty Authentication property object.
+     * @param authType     Authentication Type
+     * @param actionId     Action Id.
+     * @return Encrypted Auth Property if it is a confidential property.
+     * @throws SecretManagementException If an error occurs while encrypting the secret.
+     */
+    private AuthProperty encryptProperty(AuthProperty authProperty, String authType, String actionId)
+            throws SecretManagementException {
+
+        String secretName = buildSecretName(actionId, authType, authProperty.getName());
+        if (isSecretPropertyExists(secretName)) {
+            updateExistingSecretProperty(secretName, authProperty);
+        } else {
+            addNewActionSecretProperty(secretName, authProperty);
+        }
+
+        return new AuthProperty.AuthPropertyBuilder()
+                .name(authProperty.getName())
+                .isConfidential(authProperty.getIsConfidential())
+                .value(buildSecretReference(secretName))
+                .build();
+    }
+
+    /**
+     * Decrypt secret property.
+     *
+     * @param authProperty Authentication property object.
+     * @param authType     Authentication Type.
+     * @param actionId     Action Id.
+     * @return Decrypted Auth Property if it is a confidential property.
+     * @throws SecretManagementException If an error occurs while decrypting the secret.
+     */
+    private AuthProperty decryptProperty(AuthProperty authProperty, String authType, String actionId)
+            throws SecretManagementException {
+
+        String secretName = buildSecretName(actionId, authType, authProperty.getName());
+        if (!isSecretPropertyExists(secretName)) {
+            throw new SecretManagementException();
```

Comment:
Add a message for this exception

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5795/commits/d2420ad5643697c5521872bf43d17fe17dd12580